static/extensions/Faunks/Blobs.js

```diff
@@ -103,10 +128,43 @@
     }
 
     toBlob(args, util) {
-        const text = String(args.DATA);      // ensure it's a string (why can't it be just like python or smth)
+        const text = String(args.DATA); 
         const blob = new Blob([text], { type: args.TYPE });
         return URL.createObjectURL(blob);
-    } 
+    }
+    
+    revokeBlob(args, util){
+        URL.revokeObjectURL(args.URL);
+    }
+
+    readBlob(args, util) {
+    async function readBlobContent(url, mime) {
+        
+        if (!url.startsWith("blob:")) return ""; // I WOULD return "Url Error", but it would make it a bit harder to track if an output has been returned or not
+
+        const response = await fetch(url);
+        const blob = await response.blob();
+        
+
+        if (blob.size === 0) return ""; // js for u TheShovel
+
+        if (mime.startsWith("text/") || mime === "application/json") {
+            return await blob.text();
+        }
+
+        const buffer = await blob.arrayBuffer();
```

Comment:
After doing a bit of looking at the [MDN documentation for `Blob`](https://developer.mozilla.org/en-US/docs/Web/API/Blob), there may be a way of doing this with a readable stream, although I'd need to check how efficient that is. Also quick note that the `text` method doesn't seem to require text to actually be text, so it can pretty much be anything and still be interpreted as text.

Reply:
I could also be wrong about this.